libmscore/figuredbass.cpp

```diff
@@ -1770,6 +1773,256 @@ FiguredBass* Score::addFiguredBass()
+                              // why? because these notes were auto-generated, and their note->line() returns 0 even if not really on line 0.
+                              if ( find_if(n->chord()->notes().begin(),
+                                           n->chord()->notes().end(),
+                                           [n](Note *j) {
+                                                 // if there is a note j whose pitch is < n's pitch, then n is not the bottom note, so return.
+                                                 return j->pitch() < n->pitch();
+                                           }) != n->chord()->notes().end())
+                              {
+                                    return; // *n is not the bottom note of the chord so skip it.
+                              }
+                              if ( mod7(n->line()) != mod7(harmony_line) ) // skip if it is not same line or +/- octaves
+                                    return;
+                              if ( n->chord()->tick() > chord->tick() ) // skip if it is to the right
+                                    return;
+                              if ( n->chord()->tick() < antecedent_tick)
+                                    return;
+                              antecedent = n;
+                              antecedent_tick = n->chord()->tick();
+                        });
+                        if (antecedent_tick > -1){  // if found an antecedent.
+                              harmony = note->ppitch() + mod12(antecedent->ppitch() - note->ppitch());
+                        } else {
+                              harmony = native_pitch + int(acciv);
+                        }
+                  }
+                        break;
+                  case FiguredBassItem::Modifier::DOUBLEFLAT:
+                        harmony = native_pitch + int(AccidentalVal::FLAT2);
+                        break;
+                  case FiguredBassItem::Modifier::FLAT:
+                        harmony = native_pitch + int(AccidentalVal::FLAT);
+                        break;
+                  case FiguredBassItem::Modifier::NATURAL:
+                        harmony = native_pitch;
+                        break;
+                  case FiguredBassItem::Modifier::SHARP:
+                        harmony = native_pitch + int(AccidentalVal::SHARP);
+                        break;
+                  case FiguredBassItem::Modifier::DOUBLESHARP:
+                        harmony = native_pitch + int(AccidentalVal::SHARP);
+                        harmony = native_pitch + int(AccidentalVal(modifier));
+                        break;
+                  case FiguredBassItem::Modifier::CROSS:
+                  case FiguredBassItem::Modifier::BACKSLASH:
+                        harmony = native_pitch + int(acciv) + 1; // adding 1 semi tone, flat->natural, natural->sharp, etc.
+                        break;
```

Comment:
These details need to somehow be configurable to the end user.  What should such an interface look like?  I’m not really sure.
I imagine there should be a GUI which pops up when the Tool->Realize Figured Bass menu item gets selected.
I will need some help designing this. 


> On 24 May 2015, at 05:30, Maurizio M. Gavioli <notifications@github.com> wrote:
> 
> In libmscore/figuredbass.cpp <https://github.com/musescore/MuseScore/pull/2015#discussion_r30949847>:
> 
> > +                        harmony = native_pitch + int(AccidentalVal::FLAT);
> > +                        break;
> > +                  case FiguredBassItem::Modifier::NATURAL:
> > +                        harmony = native_pitch;
> > +                        break;
> > +                  case FiguredBassItem::Modifier::SHARP:
> > +                        harmony = native_pitch + int(AccidentalVal::SHARP);
> > +                        break;
> > +                  case FiguredBassItem::Modifier::DOUBLESHARP:
> > +                        harmony = native_pitch + int(AccidentalVal::SHARP);
> > +                        harmony = native_pitch + int(AccidentalVal(modifier));
> > +                        break;
> > +                  case FiguredBassItem::Modifier::CROSS:
> > +                  case FiguredBassItem::Modifier::BACKSLASH:
> > +                        harmony = native_pitch + int(acciv) + 1; // adding 1 semi tone, flat->natural, natural->sharp, etc.
> > +                        break;
> The meaning of the back slash varies: in some authors it mean augmented and in some diminished. Note that it does not necessarily imply a pitch modification; for instance, while in C major, a 5th in an F was commonly notated as augmented, but the unmodified B above the F has to be used.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/musescore/MuseScore/pull/2015/files#r30949847>.
>